### PR TITLE
[Refactor][tsserver] Coding Quality Improvements

### PR DESCRIPTION
**Pull request by kevinuehara**
Refator #

I'm updating some validations and improvements the code of tsserver module.

## First: `nodeServer.ts`

I change this function here, which was using switch and update to use the get object pattern

image

TO:

image

## Second: `server.ts`

I made some code improvements and avoid the repetition of same validation:

image

image

TO:

image
image

## Third: `nodeServer.ts`

I adjust this line, because when the tests was running this error of lint appears:
image

To adjust I', simple change the `null!` to `undefined`

This PR is related to the Issue that I opened: https://github.com/microsoft/TypeScript/issues/57376

### CODE REVIEW

**Comment by kevinuehara** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Comment by kevinuehara** [comment]
Somenone can help me on step `CI / format (pull_request)`?

File: src/tsserver/nodeServer.ts [hunk 1]
```diff
@@ -127,23 +135,24 @@ function parseServerMode(): LanguageServiceMode | string | undefined {
...
-        case "semantic":
-            return LanguageServiceMode.Semantic;
-        case "partialsemantic":
-            return LanguageServiceMode.PartialSemantic;
-        case "syntactic":
-            return LanguageServiceMode.Syntactic;
-        default:
-            return mode;
-    }
+    const serverMode = {
```
Comment (RyanCavanaugh) [inline src/tsserver/nodeServer.ts:138]: This incurs an object allocation on every call, and the object gets bigger every time we add more stuff. I get that this looks "cleaner" but there are reasons to use `switch` as well.

File: src/tsserver/nodeServer.ts [hunk 2]
```diff
@@ -74,7 +74,9 @@ interface NodeSocket {
     write(data: string, encoding: string): boolean;
 }
 
-function parseLoggingEnvironmentString(logEnvStr: string | undefined): LogOptions {
+function parseLoggingEnvironmentString(
```
Comment (RyanCavanaugh) [inline src/tsserver/nodeServer.ts:77]: Short signatures on one line is the project style

File: src/tsserver/nodeServer.ts [hunk 3]
```diff
@@ -717,7 +871,7 @@ function startNodeSession(options: StartSessionOptions, logger: Logger, cancella
     }
 
     const ioSession = useNodeIpc ? new IpcIOSession() : new IOSession();
-    process.on("uncaughtException", err => {
+    process.on("uncaughtException", (err) => {
```
Comment (RyanCavanaugh) [inline src/tsserver/nodeServer.ts:874]: There's nothing being fixed here, you're just changing from one style to another

**Comment by RyanCavanaugh** [comment]
There are over [2,000 issues](https://github.com/microsoft/TypeScript/issues?q=is%3Aissue+is%3Aopen+milestone%3ABacklog) in our Backlog that we're accepting PRs for. Please don't send stylistic/formatting/"cleanup" PRs that don't address an identified issue. Of all the things we could use help with, reformatting our code isn't one of them 😅

**Pull request closed without merging**